### PR TITLE
Fixes #5789.

### LINKED ISSUE #5789: Async/Await with custom promise when targeting ES6 and CommonJS
Seems like tsc uses invalid class name when targeting ES6 and CommonJS, at least with custom promises in async functions.

```typescript
// Task.ts
export class Task<T> extends Promise<T> {}

// Test.ts
import { Task } from "./Task";
class Test
{
    async Example<T>(): Task<T> {}
}
```

compiled to

```typescript
// Task.js
class Task extends Promise {}
exports.Task = Task;

// Test.js
var __awaiter = ...
var Task_1 = require("./Task");
class Test
{
    Example() { return __awaiter(this, void 0, Task, function* () {}); }
}
```

Instead of `return __awaiter(this, void 0, Task, function* () {});` should be `return __awaiter(this, void 0, Task_1.Task, function* () {});` since `Task` is imported as `Task_1.Task` when using commonjs.

1.8.0-dev.20151125
tsc --target es6 --module commonjs

### PR DESCRIPTION
**Pull request by rbuckton**
This worked previously, as the approach we were using in the past was checking the type as an expression which marked the import as referenced. That approach ran into issues with type aliases, which were addressed in a recent PR. However, there wasn't a test defined to capture this behavior. This PR fixes the issue directly and adds a test to ensure there will be no future regressions.

*Edit by @DanielRosenwasser: Fixes #5789*

### CODE REVIEW

File: tests/baselines/reference/asyncImportedPromise_es6.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+            var result = generator[verb](value);
+            result.done ? resolve(result.value) : cast(result.value).then(onfulfill, onreject);
+        }
+        step("next", void 0);
+    });
+};
+var task_1 = require("./task");
+class Test {
+    example() {
+        return __awaiter(this, void 0, Task, function* () { return; });
```
Comment (mhegazy) [inline tests/baselines/reference/asyncImportedPromise_es6.js]: should not this be `task_1.Task`?

Comment (rbuckton) [inline tests/baselines/reference/asyncImportedPromise_es6.js]: Thanks, didn't catch that one. I've pushed an update for this.

**Comment by mhegazy** [comment]
can we add a test with async getter.

**Comment by mhegazy** [comment]
and would not say no to other tests as well, with aliases and imports and internal modules as well.

**Comment by rbuckton** [comment]
@mhegazy: Async getter? We don't support that.

**Comment by rbuckton** [comment]
@mhegazy: Async getters are not in the [formal spec proposal](http://tc39.github.io/ecmascript-asyncawait/#async-function-definitions) for Async functions.

**Comment by mhegazy** [comment]
:+1:

**Comment by rbuckton** [comment]
I'll add a few more tests and merge.

**Pull request merged**